src/lib/swapper/swappers/ThorchainSwapper/utils/makeSwapMemo/assertIsValidMemo.ts

```diff
@@ -69,7 +69,14 @@ export const assertIsValidMemo = (memo: string, chainId: ChainId, affiliateBps:
   const [, pool, address, limitComponent, affiliate, memoAffiliateBps] =
     memo.split(MEMO_PART_DELIMITER)
 
-  const buyAssetChainId = thorChainAssetToChainId.get(pool.split(POOL_PART_DELIMITER)[0])
+  if (!pool) throw new Error(`pools is required in memo: ${memo}`)
+  const thorchainAsset = pool.split(POOL_PART_DELIMITER)[0]
+  if (!thorchainAsset) throw new Error(`thorchainAsset is required in memo: ${memo}`)
```

Comment:
Non-breaking change, we should always have a valid pool (i.e consisting of chain and asset parts).
Assuming we didn't on some calls:

- previous l.72 would evalute to undefined if `pool.split(POOL_PART_DELIMITER)[0]` is 
- `isAddressValid` would then be undefined 
- we would throw right under either way